openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -1250,6 +1333,48 @@ def __init__(self, **kwargs):
         super().__init__(**kwargs)
 
 
+    def check_handler_compatibility(self,
+                                    other_handler):
+        """
+        Checks whether this ParameterHandler encodes the same physics as another ParameterHandler. This is
+        called if a second handler is attempted to be initialized for the same tag.
+
+        Parameters
+        ----------
+        other_handler : a ParameterHandler-derived object
+            The handler to compare to.
+
+        Raises
+        ------
+        IncompatibleParameterError if handler_kwargs are incompatible with existing parameters.
+        """
+        float_attrs_to_compare = []
+        string_attrs_to_compare = ['potential']
```

Comment:
As opposed to splitting into different arrays, it may be a bit cleaner to have a single array of all attrs and then just do a type check in the below loops. If these checks (i.e `abs(this_val - other_val) > 1.e-6`) are going to be used repeatedly they could probably just be dumped into static methods on the `ParameterHandler` class.

Something like

```
@staticmethod
def compare_numbers(a, b, tolerance)
    ....
```

or something more like numpy's `isclose` method.

Reply:
That would be a good first step for streamlining. For now, I want to see what this function turns into. There may be some cases where differences in some attributes are permissible, so I want to develop these separately a bit more to see how much flexibility a single inherited solution would require.